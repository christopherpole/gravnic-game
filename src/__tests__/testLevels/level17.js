import { MOVE_DOWN, ENTITIES, levelIsComplete, makeMoves } from '../../index';

describe('Level 17', () => {
  it('Matches the correct snapshot', () => {
    const gameState = [
      [
        { staticEntity: null, movableEntity: null },
        {
          staticEntity: { id: 2, entityId: ENTITIES.FLOOR },
          movableEntity: { entityId: ENTITIES.BLOCK, color: '#FFFF00', id: 1 },
        },
        { staticEntity: null, movableEntity: null },
        { staticEntity: null, movableEntity: null },
      ],
      [
        {
          staticEntity: { id: 4, entityId: ENTITIES.FLOOR },
          movableEntity: { entityId: ENTITIES.SMART_BOMB, id: 3 },
        },
        { staticEntity: { id: 5, entityId: ENTITIES.FLOOR }, movableEntity: null },
        { staticEntity: null, movableEntity: null },
        {
          staticEntity: { id: 7, entityId: ENTITIES.FLOOR },
          movableEntity: { entityId: ENTITIES.BLOCK, color: '#FFFF00', id: 6 },
        },
      ],
      [
        { staticEntity: null, movableEntity: null },
        {
          staticEntity: { id: 9, entityId: ENTITIES.FLOOR },
          movableEntity: { entityId: ENTITIES.SMART_BOMB, id: 8 },
        },
        { staticEntity: null, movableEntity: null },
        { staticEntity: null, movableEntity: null },
      ],
    ];

    const gameStates = makeMoves(gameState, [MOVE_DOWN]);
    expect(gameStates).toMatchSnapshot();
    expect(
      levelIsComplete(
        gameStates[gameStates.length - 1][gameStates[gameStates.length - 1].length - 1],
      ),
    ).toBe(true);
  });
});
