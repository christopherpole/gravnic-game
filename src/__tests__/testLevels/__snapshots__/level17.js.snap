// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Level 17 Matches the correct snapshot 1`] = `
Array [
  Array [
    Array [
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 2,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
      Array [
        Object {
          "movableEntity": Object {
            "entityId": "SMART_BOMB",
            "id": 3,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 4,
          },
        },
        Object {
          "movableEntity": Object {
            "color": "#FFFF00",
            "entityId": "BLOCK",
            "id": 1,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 5,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": Object {
            "color": "#FFFF00",
            "entityId": "BLOCK",
            "id": 6,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 7,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": Object {
            "entityId": "SMART_BOMB",
            "id": 8,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 9,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
    ],
    Array [
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 2,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
      Array [
        Object {
          "movableEntity": Object {
            "entityId": "SMART_BOMB",
            "fading": true,
            "id": 3,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 4,
          },
        },
        Object {
          "movableEntity": Object {
            "color": "#FFFF00",
            "entityId": "BLOCK",
            "fading": true,
            "id": 1,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 5,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": Object {
            "color": "#FFFF00",
            "entityId": "BLOCK",
            "fading": true,
            "id": 6,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 7,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": Object {
            "entityId": "SMART_BOMB",
            "fading": true,
            "id": 8,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 9,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
    ],
    Array [
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 2,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 4,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 5,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 7,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 9,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
    ],
  ],
]
`;
